spec/component/jobs/fetch_oa_metadata_job_spec.rb

```diff
@@ -41,33 +44,40 @@
     context 'when unpaywall does not have open access information' do
       let(:unpaywall_json) { Rails.root.join('spec', 'fixtures', 'unpaywall2.json').read }
 
-      context 'when OAB has open access information' do
-        before { job.perform_now(pub.id) }
+      before { job.perform_now(pub.id) }
 
-        it 'updates publication with OAB open access information' do
-          expect(pub.reload.open_access_locations).not_to eq []
-        end
+      it 'does not update the publication oa data' do
+        expect(pub.reload.open_access_locations.where(source: Source::UNPAYWALL)).to eq []
+      end
 
-        it "updates publication's open access button last checked at" do
-          expect(pub.reload.open_access_button_last_checked_at).to be_within(1.minute).of(Time.zone.now)
-        end
+      it 'updates the publication oa status & unpaywall last checked at, oa status, and workflow state' do
+        expect(pub.reload.open_access_status).to eq 'unknown'
+        expect(pub.reload.oa_status_last_checked_at).to be_within(1.minute).of(Time.zone.now)
+        expect(pub.reload.unpaywall_last_checked_at).to be_within(1.minute).of(Time.zone.now)
+        expect(pub.reload.oa_workflow_state).to be_nil
       end
+    end
 
-      context 'when there is no open access information' do
-        before { job.perform_now(pub.id) }
+    context 'when the publication doi can be found in scholarsphere' do
+      before { job.perform_now(pub.id) }
 
-        let(:oab_json) { Rails.root.join('spec', 'fixtures', 'oab5.json').read }
+      it 'updates publication with scholarsphere open access information' do
+        expect(pub.reload.open_access_locations.where(source: Source::SCHOLARSPHERE)).not_to eq []
+      end
+    end
+
+    context 'when the publication doi cannot be found in scholarsphere' do
+      before { job.perform_now(pub.id) }
 
-        it 'does not update the publication oa data' do
-          expect(pub.reload.open_access_locations).to eq []
-          expect(pub.reload.unpaywall_last_checked_at).to be_nil
-          expect(pub.reload.open_access_button_last_checked_at).to be_nil
-        end
+      let(:ss_data) { [{ 'url' => '' }] }
+
+      it 'does not update the publication oa data' do
+        expect(pub.reload.open_access_locations.where(source: Source::SCHOLARSPHERE)).to eq []
+      end
 
-        it 'updates the publication oa status last checked at and workflow state' do
-          expect(pub.reload.oa_status_last_checked_at).to be_within(1.minute).of(Time.zone.now)
-          expect(pub.reload.oa_workflow_state).to be_nil
-        end
+      it 'updates the publication oa status & unpaywall last checked at, oa status, and workflow state' do
```

Comment:
The description of this block doesn't seem to quite match the actual assertions within. I think that the assertions are probably correct, but the way that the description reads is a little confusing to me. It looks like what we're actually asserting is that it "updates the OA status check timestamp and clears the OA workflow state". Should we be checking anything else?

Reply:
No, I think that's all that needs to be checked & you're right about just updating the assertion.